gateway/src/error.rs

```diff
@@ -400,6 +405,13 @@ impl std::fmt::Display for Error {
             Error::AzureServer { message } => {
                 write!(f, "Error from Azure server: {}", message)
             }
+            Error::BadCredentialsPreInference { provider_name } => {
+                write!(
+                    f,
+                    "Bad credentials at inference time for provider: {}. This should never happen.",
```

Comment:
For all `This should never happen` errors, append a `Please file a bug report: https://github.com/tensorzero/tensorzero/issues/new` to the message